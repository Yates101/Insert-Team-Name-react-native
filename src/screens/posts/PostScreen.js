import React, { useState, useEffect } from 'react';
import { StyleSheet, View, Button, ScrollView } from 'react-native';
import { Headbar, Post } from '../../components/index';
import axios from 'axios';
import { useFonts } from 'expo-font';
import { useNavigation } from '@react-navigation/native';

const PostList = (props) => {
  const {posts = [], user} = props
  return posts.reverse().map((post)=> <Post key={post.id} {...post} {...{user}}/>)
}

const PostsScreen = (props, { route }) => {
  const navigation = useNavigation()
  
  const [posts, setPosts] = useState([])

  const getPosts = async() => {
    await axios.get("http://localhost:3001/posts").then((res) => {
      setPosts(res.data)
    })
  }
  
  useEffect(() => {
   getPosts()
  }, [])

  return (
    <View>
      <Headbar/>
      <ScrollView>
        <View style={styles.container}>
          <PostList posts={posts} user={props.user.id}/>
        </View>
      </ScrollView>   
      <View style={styles.createPostButton}>
        <Button 
          style={styles.createPostButton} 
          title='Create Post' 
          onPress={() => navigation.navigate("Create Post")} 
        />
      </View> 
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#cdd7d6',
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
  },
  createPostButton: {
    backgroundColor: '#FFED00',
    width: "100%",
    height: 66,
    borderRadius: 33,
    justifyContent: 'center',
    alignItems:'center',
    position: 'absolute',
    bottom: 50,
  }
});

export {PostsScreen};