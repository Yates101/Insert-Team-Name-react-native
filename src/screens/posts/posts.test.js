import 'react-native';
import React from 'react';

import {  PostsScreen } from './PostScreen.js';
import renderer from 'react-test-renderer';
import '@testing-library/jest-native/extend-expect';

describe("< PostsScreen/>", () => {
  const tree = renderer.create(< PostsScreen />).toJSON();

  it('has 3 children', () => {
    expect(tree.children.length).toBe(3);
  });

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  })
})
