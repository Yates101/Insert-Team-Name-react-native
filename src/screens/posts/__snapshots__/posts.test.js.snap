// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< PostsScreen/> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "alignSelf": "stretch",
        "backgroundColor": "#179aff",
        "flexDirection": "row",
        "height": 52,
        "justifyContent": "space-between",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFED00",
          "fontSize": 32,
          "fontWeight": "bold",
        }
      }
    >
      IkeBook
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#cdd7d6",
            "height": "100%",
            "justifyContent": "center",
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFED00",
        "borderRadius": 33,
        "bottom": 50,
        "height": 66,
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Create Post
        </Text>
      </View>
    </View>
  </View>
</View>
`;
